### ISSUE TITLE
[typescript-operations] it should be possible to set avoidOptionals only for input types

### ISSUE THREAD

**Issue by helios1138** [dotansimha/graphql-code-generator#1885]
- currently, it's only possible to set `avoidOptionals` globally, for all types
- what would be preferred, is to have a way to set this option to `true` for output types (since empty values are returned as `null`, not `undefined` in the responses), but set it to `false` for input types (since variables treat `undefined` as null and it's a normal practice to omit empty values when providing variables, rather than providing `null`s

**Comment by dotansimha** [comment]
Hi @yoavain-sundaysky ! Thank you for reporting this issue.
You are right, at the moment the optionals configuration can only be turned on/off and is not configurable.
We can give it a try in the future and add it (and we always welcome PRs).

**Comment by slikts** [comment]
An example use case for `avoidOptionals` is to model mock data in strictly the same way the server would return it, i.e., with `null` for requested nullable fields. Meanwhile, nullable fields in variable and input types should still be optional, because being able to elide the fields is also how it works in GraphQL, and having to always explicitly provide a `null` or `[]` is verbose and poor DX.

A workaround to currently avoid using `avoidOptionals` is a helper that removes optionality:
```ts
type DeepNonNullable<T> = {
  [P in keyof T]-?: T[P] extends {} ? DeepNonNullable<T[P]> : T[P];
}
```

**Comment by dotansimha** [comment]
Implemented in https://github.com/dotansimha/graphql-code-generator/pull/2191 . Thanks @ekfn

**Comment by dotansimha** [comment]
Available in 1.5.0 🎉

**Comment by thefliik** [comment]
@dotansimha it doesn't look like this issue should be closed yet. In my testing, #2191 fixes this issue for the `typescript` plugin, but not for the `typescript-operations` plugin.

It appears that `typescript-operations` still only accepts a boolean value for `avoidOptionals`. Attempting to pass an object value is treated the same as passing `avoidOptionals: true`.

**Comment by dotansimha** [comment]
@thefliik you are right, I added a validation for the type of `avoidOptionals` in `typescript-operations`:
```
        avoidOptionals: typeof config.avoidOptionals === 'boolean' ? getConfigValue(config.avoidOptionals, false) : false,

```

Now it will ignore the object value if it set globally.

**Comment by dotansimha** [comment]
Fixed in 1.6.0
